### ISSUE TITLE
[data grid] Pinned columns and rows in Data Grid Pro fail accessibility test

### ISSUE THREAD

**Issue by DavidArmendariz** [mui/mui-x#5364]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

We are using jest-axe for our accessibility tests. Currently, we use the Data Grid Pro with the pinned columns feature to pin one column to the left and one to the right. Jest-axe complains that these columns have the "columnheader" role but their parent components (which we don't have any control) need to have the "row" role.

```
Expected the HTML found at $('.MuiDataGrid-pinnedColumnHeaders--left > .MuiDataGrid-columnHeader.MuiDataGrid-columnHeader--sortable[role="columnheader"]') to have no violations:

<div class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable" data-field="col1" style="width: 150px; min-width: 150px; max-width: 150px;" role="columnheader" tabindex="0" aria-colindex="1" aria-sort="none">

Received:

"Certain ARIA roles must be contained by particular parents (aria-required-parent)"

Fix any of the following:
  Required ARIA parent role not present: row

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.4/aria-required-parent?application=axeAPI
```

Also, when the data grid is not empty, every row throws an accessibility problem like this:

```
Expected the HTML found at $('.MuiDataGrid-virtualScrollerRenderZone > div[data-id="1"][data-rowindex="0"][aria-rowindex="2"]') to have no violations:

<div data-id="1" data-rowindex="0" role="row" class="MuiDataGrid-row" aria-rowindex="2" aria-selected="false" style="max-height: 52px; min-height: 52px;"></div>

Received:

"Certain ARIA roles must contain particular children (aria-required-children)"

Fix any of the following:
  Required ARIA children role not present: cell, columnheader, gridcell, rowheader

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.4/aria-required-children?application=axeAPI
```

### Expected behavior 🤔

The expected behavior would be that the parent of the columns have the "row" role and that the children of the rows have the correct roles to avoid the "aria-required-children" problem. If there is a reason this was intentionally built like this, we would like to know why.

### Steps to reproduce 🕹

Codesandbox link: https://codesandbox.io/s/data-grid-pro-forked-46hj83?file=/src/App.tsx

Steps:

1. Create a Data Grid Pro with any columns and any data (disable virtualization for testing purposes as suggested [in the documentation](https://github.com/mui/mui-x/issues/5364))
2. Pin any column to the left or to the right or both
3. Create a simple test with jest-axe as shown in [their documentation](https://github.com/nickcolley/jest-axe)
4. Confirm all the accessibility problems thrown by jest-axe

### Context 🔦

We try that all of our components are accessible and for that we use jest-axe.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
System:
    OS: Linux 5.13 Ubuntu 20.04.4 LTS (Focal Fossa)
  Binaries:
    Node: 17.9.0 - ~/.nvm/versions/node/v17.9.0/bin/node
    Yarn: 1.22.18 - ~/.nvm/versions/node/v17.9.0/bin/yarn
    npm: 8.5.5 - ~/.nvm/versions/node/v17.9.0/bin/npm
  Browsers:
    Chrome: 103.0.5060.53
    Firefox: 101.0.1
  npmPackages:
    @mui/base:  5.0.0-alpha.87 
    @mui/material: ^5.8.6 => 5.8.6 
    @mui/private-theming:  5.8.6 
    @mui/styled-engine-sc:  5.4.2 
    @mui/system: ^5.8.6 => 5.8.6 
    @mui/types:  7.1.4 
    @mui/utils:  5.8.6 
    @mui/x-data-grid:  5.12.3 
    @mui/x-data-grid-pro: ^5.12.3 => 5.12.3 
    @mui/x-license-pro: ^5.12.3 => 5.12.3 
    @types/react: ^17.0.40 => 17.0.40 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2 
    styled-components: ^5.3.5 => 5.3.5 
    typescript: ^4.7.4 => 4.7.4
```
</details>

### Order ID 💳 (optional)

46207
